### PR TITLE
Add upgrade guide 3.11 to 3.12

### PR DESCRIPTION
**Pull request by kadewu**
Migration upgrade guide. To be merged after succesfull resolving all PR's from issue: https://github.com/saleor/saleor/issues/11337

### CODE REVIEW

File: docs/upgrade-guides/3-11-to-3-12.mdx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: Upgrading from 3.11.x to 3.12
+---
+
+## Preparation
+
+Version 3.12 introduces new fields in our model. To properly migrate from to 3.12 please migrate first to version 3.11.4
```
Comment (peelar) [inline docs/upgrade-guides/3-11-to-3-12.mdx:7]:
```suggestion
Version 3.12 introduces new fields in our model. To successfully upgrade from version 3.11.x to 3.12, please migrate to 3.11.4 first.
```

File: docs/upgrade-guides/3-11-to-3-12.mdx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+## Reasoning
+
+In version 3.12 `orderSettings` were moved to `Channel`. For more info relate to https://github.com/saleor/saleor/blob/main/CHANGELOG.md
```
Comment (peelar) [inline docs/upgrade-guides/3-11-to-3-12.mdx:11]:
```suggestion
For more info relate to https://github.com/saleor/saleor/blob/main/CHANGELOG.md
```

Three points here:
- please make sure to include the "." at the end of the sentence
- please format the links, e.g. `[Changelog](https://github.com/saleor/saleor/blob/main/CHANGELOG.md)`
- the sentence should probably be something like "for more related information, please...", but...

...I think the documentation should be the source of truth here. Changelog is a nice addition, but we should at least briefly inform about the consequences of the appointed changes.

**Review by peelar — CHANGES_REQUESTED** [review]

**Comment by peelar** [comment]
You can have a look at [the writing documentation guidelines](https://github.com/saleor/saleor-docs/blob/main/GUIDELINES.md) I just merged.

File: docs/upgrade-guides/3-11-to-3-12.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (korycins) [reply docs/upgrade-guides/3-11-to-3-12.mdx:7]: I think that we could be more specific than `our model`. Also, for someone who is not familiar with Django, the meaning of `model` can be not fully understood. 
Also, I think, that the preparation section should be more generic. I am going to add a migration guide related to the transaction and changes in the preview feature part, and the current preparation content looks like something strictly related to the channel changes.
Would be also good to mention, that the mentioned request to bump to 3.11.4 first, is strictly related to the zero downtime upgrade. This step can be skipped by someone who doesn't care about it. Or he doesn't use Saleor in production yet.

**Comment by peelar** [comment]
@IKarbowiak I merged your changes here.

**Comment by peelar** [comment]
@IKarbowiak Please provide a link to an example API Reference object field that will be affected by these changes.

**Comment by IKarbowiak** [comment]
@peelar done :)

**Review by peelar — APPROVED** [review]

**Comment by peelar** [comment]
@korycins will you be reviewing?

**Review by korycins — APPROVED** [review]

**Pull request merged**